src/rpc/parsers.ts

```diff
@@ -139,7 +139,18 @@ function parseSuccessful(
             : xdr.ScVal.scvVoid()
         };
       })[0]
-    })
+    }),
+
+    stateChanges: sim.stateChanges?.length ?? 0 > 0
```

Comment:
Doing it this way means `stateChanges` is always present, and might be set as `stateChanges: undefined`. We want to omit the property altogether. This is why we need the spread operator. Observe the differences:

```js
$ node         
> {
  value: [].length > 0 ? "hello" : undefined
}
{ value: undefined }
> // vs.
> {
  ...([].length > 0 ?? { value: "hello" })
}
{}
```

Reply:
Yes I think we are on same page for `stateChanges`, I thought you were proposing the same for `before` and `after` property which seemed wrong but gotchu now :)

Reply:
I was but I changed my mind - we should stick to the RPC API :+1: